### PR TITLE
Improve external dependency discovery logic

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
Packages can have few different layouts, the logic to detect if a dep is an external dep was not handling all the cases.

### CODE REVIEW

File: Fixtures/DependencyResolution/External/CUsingCDep2/Bar/Sources/SeaLover/Sea.c [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include <Foo/Foo.h>
+
+void cool() {
+    foo();
+}
```
Comment (Danappelxx) [inline Fixtures/DependencyResolution/External/CUsingCDep2/Bar/Sources/SeaLover/Sea.c]: Very minor thing but this is missing a line break at the end of the file :)

**Comment by mxcl** [comment]
Really, we shouldn't be detecting this, we should specify explicitly external deps at a higher level.

**Comment by aciidb0mb3r** [comment]
Yeah I understand this is a total hack to begin with, I can try fixing it next week.

**Comment by aciidb0mb3r** [comment]
@mxcl Fixed, Review/CI

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by aciidb0mb3r** [comment]
fixed the linux failure, thanks @swift-ci 🖖
@mxcl can you rerun CI

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by mxcl** [comment]
LGTM, please merge when you're done.

**Pull request merged**